blocklist-client/src/client/sanctions.rs

```diff
@@ -0,0 +1,308 @@
+use crate::config::RiskAnalysisConfig;
+use reqwest::{Client, Response, StatusCode};
+use serde::Deserialize;
+use std::error::Error as StdError;
+use tracing::debug;
+const API_BASE_PATH: &str = "/api/v1/address";
+
+/// Represents the identification information for a blockchain address.
+#[derive(Debug, Deserialize)]
+pub struct Identification {
+    /// The Chainalysis Entity category. For sanctioned addresses, the value will be 'sanctions'.
+    #[serde(rename = "category")]
+    pub _category: String,
+    /// The OFAC name associated with the sanctioned address.
+    #[serde(rename = "name")]
+    pub _name: Option<String>,
+    /// The OFAC description of the sanctioned address.
+    #[serde(rename = "description")]
+    pub _description: Option<String>,
+    /// The OFAC URL for more information about the sanctioned address.
+    #[serde(rename = "url")]
+    pub _url: Option<String>,
+}
+
+/// Structure of responce of the sanctions API
+#[derive(Debug, Deserialize)]
+pub struct SanctionsResponce {
+    /// Array with sanctions data. If empty, the address is not blocklisted.
+    pub identifications: Vec<Identification>,
+}
+
+fn risk_assessment_path(base_url: &str, address: &str) -> String {
+    format!("{}{}/{}", base_url, API_BASE_PATH, address)
+}
+
+/// Check risk status associated with a registered address
+async fn get_risk_assessment(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<RiskAssessment, Error> {
+    let api_url = risk_assessment_path(&config.api_url, address);
+    debug!("Beginning risk assessment for address: {address}");
+    let response = client
+        .get(&api_url)
+        .header("X-API-Key", &config.api_key)
+        .header("Accept", "application/json")
+        .send()
+        .await?;
+    let checked_response = check_api_response(response).await?;
+    let resp_result = checked_response.json::<SanctionsResponce>().await;
+    match resp_result {
+        Ok(resp) => {
+            if resp.identifications.is_empty() {
+                Ok(RiskAssessment {
+                    severity: RiskSeverity::Low,
+                    reason: None,
+                })
+            } else {
+                Ok(RiskAssessment {
+                    severity: RiskSeverity::Severe,
+                    reason: Some("sanctions".to_string()),
+                })
+            }
+        }
+        Err(e) if e.is_decode() => {
+            // Check if the source of the error is serde_json::Error
+            if let Some(serde_err) = e
+                .source()
+                .and_then(|cause| cause.downcast_ref::<serde_json::Error>())
+            {
+                match serde_err.classify() {
+                    serde_json::error::Category::Data => Err(Error::InvalidApiResponse),
+                    _ => Err(Error::Serialization(serde_err.to_string())),
+                }
+            } else {
+                Err(Error::Network(e))
+            }
+        }
```

Comment:
This error handling looks good! I'm realizing that throwing any error from the blocklist client will then require the sbtc signer to decide what to do in response to the error, but I'd rather leave that kind logic out of the sbtc signer.

What do you think of adding a config parameter that, when set to true, the blocklist client will reject an address if there's an internal error processing it, and when it's set to false it will always approve on an error? We would definitely want the default behavior to be a rejection, but setting it to true could be useful for testing.

Reply:
That would be a new unplanned ticket, but it'll be easier to execute well than adding the logic to the sbtc signer.

Reply:
Once you've had a chance to make the issue could you post the link here? I'll then resolve this conversation :)